metadata-ingestion/scripts/generate_lineage_doc.py

```diff
@@ -0,0 +1,210 @@
+Please refer to our [UI Guides on Lineage](../../features/feature-guides/ui-lineage.md) for more information.
+
+:::caution Recommendation on UI-based lineage
+
+Lineage added by hand and programmatically may conflict with one another to cause unwanted overwrites.
+It is strongly recommend that lineage is edited manually in cases where lineage information is not also extracted in automated fashion, e.g. by running an ingestion source.
+
+:::
+
+### API
+
+If you are not using a Lineage-support ingestion source, you can programmatically emit lineage edges between entities via API.
+Please refer to [API Guides on Lineage](../../api/tutorials/lineage.md) for more information.
+
+
+## Lineage Support
+
+DataHub supports **[automatic table- and column-level lineage detection](#automatic-lineage-extraction-support)** from BigQuery, Snowflake, dbt, Looker, PowerBI, and 20+ modern data tools. 
+For data tools with limited native lineage tracking, [**DataHub's SQL Parser**](../../lineage/sql_parsing.md) detects lineage with 97-99% accuracy, ensuring teams will have high quality lineage graphs across all corners of their data stack.
+
+### Types of Lineage Connections
+
+Types of lineage connections supported in DataHub and the example codes are as follows.
+
+* Dataset to Dataset
+    * [Dataset Lineage](../../../metadata-ingestion/examples/library/lineage_emitter_rest.py)
+    * [Finegrained Dataset Lineage](../../../metadata-ingestion/examples/library/lineage_emitter_dataset_finegrained.py)
+    * [Datahub BigQuery Lineage](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/snowflake.py#L249)
+    * [Dataset Lineage via MCPW REST Emitter](../../../metadata-ingestion/examples/library/lineage_emitter_mcpw_rest.py)
+    * [Dataset Lineage via Kafka Emitter](../../../metadata-ingestion/examples/library/lineage_emitter_kafka.py)
+* [DataJob to DataFlow](../../../metadata-ingestion/examples/library/lineage_job_dataflow.py)
+* [DataJob to Dataset](../../../metadata-ingestion/examples/library/lineage_dataset_job_dataset.py)
+* [Chart to Dashboard](../../../metadata-ingestion/examples/library/lineage_chart_dashboard.py)
+* [Chart to Dataset](../../../metadata-ingestion/examples/library/lineage_dataset_chart.py)
+
+### Automatic Lineage Extraction Support
+
+This is a summary of automatic lineage extraciton support in our data source. Please refer to the **Important Capabilities** table in the source documentation. Note that even if the source does not support automatic extraction, you can still add lineage manually using our API & SDKs.\n"""
+        )
+
+        f.write(
+            "\n| Source | Table-Level Lineage | Column-Level Lineage | Related Configs |\n"
+        )
+        f.write("| ---------- | ------ | ----- |----- |\n")
+
+        for platform_id, platform in platforms.items():
```

Comment:
at some point, I want to start migrating this stuff to jinja templating or something similar. embedding a bunch of markdown inside python code isn't great

that said, moving it to a separate file does help readability - so it's a step in the right direction